draft-ietf-mls-protocol.md

```diff
@@ -2536,7 +2561,7 @@ the right.
 * Validate the KeyPackage:
 
     * Verify that the signature on the KeyPackage is valid using the public key
-      in the KeyPackage's credential
+      in the KeyPackage's credential and the label `"KeyPackage"`
```

Comment:
I would prefer that we put these descriptions with the structs they belong to, as opposed to down here in the instructions.  We already have prose that says what the signature covers; that should just specify the label as well.  You could also include it as a comment in the syntax specification:

```
struct {
    ...
    // SignWithLabel(., "MLSPlaintextTBS", MLSPlaintextTBS)
    opaque signature<0..2^16-1>;
} KeyPackage;

struct {
    ...
    // SignWithLabel(., "GroupInfoTBS", GroupInfoTBS)
    opaque signature<0..2^16-1>;
} GroupInfo;

struct {
    ...
    // SignWithLabel(., "PublicGroupStateTBS", PublicGroupStateTBS)
    opaque signature<0..2^16-1>;
} PublicGroupState;

struct {
    ...
    // SignWithLabel(., "MLSPlaintextTBS", MLSPlaintextTBS)
    opaque signature<0..2^16-1>;

    // MAC(confirmation_key, GroupContext.confirmed_transcript_hash)
    optional<MAC> confirmation_tag;

    // MAC(membership_key, MLSPlaintextTBM);
    optional<MAC> membership_tag;
} MLSPlaintext;
```

(Note: I don't think GroupInfoTBS exists, but it probably should.)

Reply:
(#529 adds a `GroupInfoTBS`)

Reply:
If every structure has a `TBS` associated with it, this seems like a nice formatting of how you must compute the signatures, however for `KeyPackage` there is no `TBS` (yet), it is said in the text above that the TBS contains "all of the fields except for the signature field".

Should we add a `TBS` for every structure that we sign? This is for example done for `PublicGroupState` where a `TBS` is explicitly described even if it contains all fields except the signature field in `PublicGroupState`.